src/Riok.Mapperly/Descriptors/Mappings/TypeMapping.cs

```diff
@@ -12,12 +13,22 @@ protected TypeMapping(ITypeSymbol sourceType, ITypeSymbol targetType)
     {
         SourceType = sourceType;
         TargetType = targetType;
+        Parameters = ImmutableEquatableArray<MethodParameter>.Empty;
+    }
+
+    protected TypeMapping(ITypeSymbol sourceType, ITypeSymbol targetType, ImmutableEquatableArray<MethodParameter> parameters)
+    {
+        SourceType = sourceType;
+        TargetType = targetType;
+        Parameters = parameters;
     }
 
     public ITypeSymbol SourceType { get; }
 
     public ITypeSymbol TargetType { get; }
 
+    public ImmutableEquatableArray<MethodParameter> Parameters { get; }
```

Comment:
With the approach that we never pass parameters to other methods, only user defined mapping methods should not support parameters. For eg. user implemented mappings a diagnostic should be emitted if additional parameters are provided (RMG001).